grpc/ra-wrappers.go

```diff
@@ -224,14 +224,16 @@ func (ras *RegistrationAuthorityServerWrapper) NewAuthorization(ctx context.Cont
 }
 
 func (ras *RegistrationAuthorityServerWrapper) NewCertificate(ctx context.Context, request *rapb.NewCertificateRequest) (*corepb.Certificate, error) {
-	if request == nil || request.Csr == nil || request.RegID == 0 {
+	// Because this method is APIv1-only, the IssuerNameID is required so the CA
+	// never has to guess on the issuer for v1 issuance.
+	if request == nil || request.Csr == nil || request.RegID == 0 || request.IssuerNameID == 0 {
```

Comment:
I think this breaks deployability guidelines because the WFE will be sending NewCertificate RPCs with IssuerNameID unset, right?

Reply:
Yep, you're right. Turned this into a TODO instead.